diff -Nru whoson-1.09/Makefile.am whoson-1.09.new/Makefile.am
--- whoson-1.09/Makefile.am	Tue Aug 31 19:24:58 1999
+++ whoson-1.09.new/Makefile.am	Tue Aug 31 19:25:11 1999
@@ -13,55 +13,54 @@
 CLNTSOURCES = clnt_tcp.c clnt_unix.c clnt_unixd.c clnt_udp.c
 SERVSOURCES = serv_tcp.c serv_unix.c serv_unixd.c serv_udp.c
 
-bin_PROGRAMS = wtest
-libexec_PROGRAMS = whosond
-lib_LIBRARIES = libwhoson.a
-lib_LTLIBRARIES = libwhoson.la
-include_INCLUDES = whoson.h
-man_MANS = whoson.conf.5 whoson.3 whosond.8
+bin_PROGRAMS =		wtest
+libexec_PROGRAMS =	whosond
+lib_LIBRARIES =		libwhoson.a
+lib_LTLIBRARIES =	libwhoson.la
 
-EXTRA_DIST = whoson.ms whoson.txt fix.pl whoson.conf $(man_MANS) \
+man_MANS =	whoson.conf.5 whosond.8 \
+		whoson.3 wso_version.3 wso_login.3 wso_logout.3 wso_query.3
+
+EXTRA_DIST =	whoson.ms whoson.txt fix.pl \
 		WHOSON/Changes WHOSON/MANIFEST WHOSON/Makefile.PL \
 		WHOSON/WHOSON.pm WHOSON/WHOSON.xs WHOSON/test.pl
 
-wtest_SOURCES = wtest.c
-wtest_LDADD = libwhoson.la
-wtest_LDFLAGS = -static
+wtest_SOURCES =	wtest.c
+wtest_LDADD =	libwhoson.la
+wtest_LDFLAGS =	-static
 
-whosond_SOURCES = serv_common.c whosond.c mainloop.c rtconfig.c servlist.c \
+whosond_SOURCES =	serv_common.c whosond.c mainloop.c rtconfig.c servlist.c \
 			lhash.c checkperm.c ${SERVSOURCES} \
 			whosond.h rtconfig.h report.h serv_common.h \
 			lhash.h checkperm.h
 
-libwhoson_a_SOURCES = clnt_common.c rtconfig.c clntlist.c checkperm.c \
+whosond_conf =		whoson.conf
+DEFAULT_CONFIG =	$(sysconfdir)/$(whosond_conf)
+
+libwhoson_a_SOURCES =	clnt_common.c rtconfig.c clntlist.c checkperm.c \
 			${CLNTSOURCES} \
-			whoson.h rtconfig.h report.h clnt_common.h \
+			rtconfig.h report.h clnt_common.h \
 			lhash.h checkperm.h
 
-libwhoson_la_SOURCES = clnt_common.c rtconfig.c clntlist.c checkperm.c \
+libwhoson_la_SOURCES =	clnt_common.c rtconfig.c clntlist.c checkperm.c \
 			${CLNTSOURCES} \
-			whoson.h rtconfig.h report.h clnt_common.h \
+			rtconfig.h report.h clnt_common.h \
 			lhash.h checkperm.h
-libwhoson_la_LDFLAGS = -version-info 0:0:0
+libwhoson_la_LDFLAGS =	-version-info 0:0:0
 
-MAN3XTRA = wso_version.3 wso_login.3 wso_logout.3 wso_query.3
+libwhosonincludedir=		$(includedir)
+libwhosoninclude_HEADERS =	whoson.h
 
 install-data-hook:
-	if test -f $(DEFAULT_CONFIG); then \
-		$(ECHO) "$(DEFAULT_CONFIG) exists, not overwritten"; \
+	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)
+	if test -f $(DESTDIR)$(DEFAULT_CONFIG); then \
+		$(ECHO) "$(DESTDIR)$(DEFAULT_CONFIG) exists, not overwritten"; \
 	else \
-		$(INSTALL_DATA) $(srcdir)/whoson.conf $(DEFAULT_CONFIG); \
-	fi; \
-	for f in $(MAN3XTRA); do \
-		rm -f $(DESTDIR)$(man3dir)/$$f; \
-		$(LN) $(DESTDIR)$(man3dir)/whoson.3 $(DESTDIR)$(man3dir)/$$f; \
-	done
+		$(INSTALL_DATA) $(srcdir)/whoson.conf $(DESTDIR)$(DEFAULT_CONFIG); \
+	fi
 
 uninstall-local:
-	rm -f $(DEFAULT_CONFIG); \
-	for f in $(MAN3XTRA); do \
-		rm -f $(DESTDIR)$(man3dir)/$$f; \
-	done
+	rm -f $(DESTDIR)$(DEFAULT_CONFIG)
 
 whoson.txt: whoson.ms
 	${NROFF} -ms ${srcdir}/whoson.ms | ${PERL} ${srcdir}/fix.pl >whoson.txt
diff -Nru whoson-1.09/aclocal.m4 whoson-1.09.new/aclocal.m4
--- whoson-1.09/aclocal.m4	Tue Aug 31 19:24:58 1999
+++ whoson-1.09.new/aclocal.m4	Tue Aug 31 19:24:25 1999
@@ -1,4 +1,4 @@
-dnl aclocal.m4 generated automatically by aclocal 1.4
+dnl aclocal.m4 generated automatically by aclocal 1.4a
 
 dnl Copyright (C) 1994, 1995-8, 1999 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
@@ -21,6 +21,8 @@
 
 AC_DEFUN(AM_INIT_AUTOMAKE,
 [AC_REQUIRE([AC_PROG_INSTALL])
+dnl We require 2.13 because we rely on SHELL being computed by configure.
+AC_PREREQ([2.13])
 PACKAGE=[$1]
 AC_SUBST(PACKAGE)
 VERSION=[$2]
@@ -172,7 +174,12 @@
 dnl
 
 # Check for any special flags to pass to ltconfig.
-libtool_flags="--cache-file=$cache_file"
+#
+# the following will cause an existing older ltconfig to fail, so
+# we ignore this at the expense of the cache file... Checking this 
+# will just take longer ... bummer!
+#libtool_flags="--cache-file=$cache_file"
+#
 test "$enable_shared" = no && libtool_flags="$libtool_flags --disable-shared"
 test "$enable_static" = no && libtool_flags="$libtool_flags --disable-static"
 test "$enable_fast_install" = no && libtool_flags="$libtool_flags --disable-fast-install"
diff -Nru whoson-1.09/config.h.in whoson-1.09.new/config.h.in
--- whoson-1.09/config.h.in	Tue Aug 31 19:24:58 1999
+++ whoson-1.09.new/config.h.in	Tue Aug 31 19:24:25 1999
@@ -15,9 +15,6 @@
 /* Define to `int' if <sys/types.h> doesn't define.  */
 #undef uid_t
 
-/* Default config file path */
-#undef DEFAULT_CONFIG
-
  
 /* Length of unix domain socket path */
 #undef UNIX_PATH_MAX
diff -Nru whoson-1.09/configure.in whoson-1.09.new/configure.in
--- whoson-1.09/configure.in	Tue Aug 31 19:24:58 1999
+++ whoson-1.09.new/configure.in	Tue Aug 31 19:24:25 1999
@@ -48,15 +48,6 @@
 	CFLAGS="$CFLAGS -Wall"
 fi
 
-AC_ARG_WITH(config, [  --with-config=LOC       location for the config file (default /etc/whoson.conf)],
-        use_config=1)
-if test "X$with_config" != "Xno" && test "X$with_config" != "Xyes" && test "X$with_config" != "X"; then
-	DEFAULT_CONFIG=$with_config
-fi
-: ${DEFAULT_CONFIG=/etc/whoson.conf}
-AC_SUBST(DEFAULT_CONFIG)dnl
-AC_DEFINE_UNQUOTED(DEFAULT_CONFIG,"$DEFAULT_CONFIG")dnl
-
 AC_MSG_RESULT([** Using C compiler: $CC])
 AC_MSG_RESULT([** Using CFLAGS:     $CFLAGS])
 AC_MSG_RESULT([** Using config:     $DEFAULT_CONFIG])
diff -Nru whoson-1.09/wso_login.3 whoson-1.09.new/wso_login.3
--- whoson-1.09/wso_login.3	Thu Jan  1 01:00:00 1970
+++ whoson-1.09.new/wso_login.3	Tue Aug 31 19:24:25 1999
@@ -0,0 +1 @@
+.so whoson.3
diff -Nru whoson-1.09/wso_logout.3 whoson-1.09.new/wso_logout.3
--- whoson-1.09/wso_logout.3	Thu Jan  1 01:00:00 1970
+++ whoson-1.09.new/wso_logout.3	Tue Aug 31 19:24:25 1999
@@ -0,0 +1 @@
+.so whoson.3
diff -Nru whoson-1.09/wso_query.3 whoson-1.09.new/wso_query.3
--- whoson-1.09/wso_query.3	Thu Jan  1 01:00:00 1970
+++ whoson-1.09.new/wso_query.3	Tue Aug 31 19:24:25 1999
@@ -0,0 +1 @@
+.so whoson.3
diff -Nru whoson-1.09/wso_version.3 whoson-1.09.new/wso_version.3
--- whoson-1.09/wso_version.3	Thu Jan  1 01:00:00 1970
+++ whoson-1.09.new/wso_version.3	Tue Aug 31 19:24:25 1999
@@ -0,0 +1 @@
+.so whoson.3
